src/common/hotbar-store.ts

```diff
@@ -8,6 +8,8 @@ import isNull from "lodash/isNull";
 export interface HotbarItem {
   entity: {
     uid: string;
+    name: string;
+    source: string;
```

Comment:
Those extra data needed for disabled HotbarIcon, since entity is might be disappeared, but we still need to show name and background color for `disabled` item.

Reply:
Have you tested the flow of changing an entity's name? Does the hotbar icon update accordingly?